src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -22,147 +23,160 @@ internal class ListPackageCommandRunner : IListPackageCommandRunner
             }
             //If the given file is a solution, get the list of projects
             //If not, then it's a project, which is put in a list
             var projectsPaths = Path.GetExtension(listPackageArgs.Path).Equals(".sln") ?
                            MSBuildAPIUtility.GetProjectsFromSolution(listPackageArgs.Path).Where(f => File.Exists(f)) :
                            new List<string>(new string[] { listPackageArgs.Path });
 
-            var autoReferenceFound = false;
-            var msBuild = new MSBuildAPIUtility(listPackageArgs.Logger);
+            MSBuildAPIUtility msBuild = listPackageReportModel.MSBuildAPIUtility;
 
-            //Print sources, but not for generic list (which is offline)
-            if (listPackageArgs.ReportType != ReportType.Default)
+            foreach (string projectPath in projectsPaths)
             {
-                Console.WriteLine();
-                Console.WriteLine(Strings.ListPkg_SourcesUsedDescription);
-                ProjectPackagesPrintUtility.PrintSources(listPackageArgs.PackageSources);
-                Console.WriteLine();
+                await GetProjectMetadataAsync(projectPath, listPackageReportModel, msBuild, listPackageArgs);
             }
 
-            foreach (var projectPath in projectsPaths)
-            {
-                //Open project to evaluate properties for the assets
-                //file and the name of the project
-                var project = MSBuildAPIUtility.GetProject(projectPath);
+            // if there is any error then return failure code.
+            int exitCode = (
+                listPackageArgs.Renderer.GetProblems().Any(p => p.ProblemType == ProblemType.Error)
+                || listPackageReportModel.Projects.Where(p => p.ProjectProblems != null).SelectMany(p => p.ProjectProblems).Any(p => p.ProblemType == ProblemType.Error))
+                ? GenericFailureExitCode : GenericSuccessExitCode;
 
-                if (!MSBuildAPIUtility.IsPackageReferenceProject(project))
-                {
-                    Console.Error.WriteLine(string.Format(CultureInfo.CurrentCulture,
-                        Strings.Error_NotPRProject,
-                        projectPath));
-                    Console.WriteLine();
-                    continue;
-                }
+            return (exitCode, listPackageReportModel);
+        }
+
+        private async Task GetProjectMetadataAsync(
+            string projectPath,
+            ListPackageReportModel listPackageReportModel,
+            MSBuildAPIUtility msBuild,
+            ListPackageArgs listPackageArgs)
+        {
+            //Open project to evaluate properties for the assets
+            //file and the name of the project
+            Project project = MSBuildAPIUtility.GetProject(projectPath);
+            var projectName = project.GetPropertyValue(ProjectName);
+            // Project specific data stored in below variable
```

Comment:
These comments are repeating what the code does, so I don't think it adds value.